code/game/machinery/lightswitch.dm

```diff
@@ -82,3 +91,41 @@
 		return
 	if(!(machine_stat & (BROKEN|NOPOWER)))
 		power_change()
+
+/obj/item/circuit_component/light_switch
+	display_name = "Light Switch"
+	desc = "Allows to control the lights of an area."
+	circuit_flags = CIRCUIT_FLAG_INPUT_SIGNAL
```

Comment:
Not a fan of this, since this'll just give "Trigger". I think you should make an explicit input port called "Toggle", as well as "Turn On" and "Turn Off".

Reply:
though I think you should leave the input signal in too, because there's no telling if the light switch is on or off yet.

Reply:
basically, have an extra port to determine if it gets switched on or off and use the trigger port to make it do so.